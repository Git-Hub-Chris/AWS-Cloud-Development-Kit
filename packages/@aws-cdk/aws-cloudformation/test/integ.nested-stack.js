"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const lambda = require("@aws-cdk/aws-lambda");
const sns = require("@aws-cdk/aws-sns");
const sns_subscriptions = require("@aws-cdk/aws-sns-subscriptions");
const sqs = require("@aws-cdk/aws-sqs");
const core_1 = require("@aws-cdk/core");
class MyNestedStack extends core_1.NestedStack {
    constructor(scope, id, props) {
        const topicNamePrefixLogicalId = 'TopicNamePrefix';
        super(scope, id, {
            parameters: {
                [topicNamePrefixLogicalId]: props.topicNamePrefix,
            },
            description: props.description,
        });
        const topicNamePrefixParameter = new core_1.CfnParameter(this, 'TopicNamePrefix', { type: 'String' });
        for (let i = 0; i < props.topicCount; ++i) {
            const topic = new sns.Topic(this, `topic-${i}`, { displayName: `${topicNamePrefixParameter.valueAsString}-${i}` });
            // since the subscription resources are defined in the subscriber's stack, this
            // will add an SNS subscription resource to the parent stack that reference this topic.
            if (props.subscriber) {
                topic.addSubscription(new sns_subscriptions.SqsSubscription(props.subscriber));
            }
        }
        if (props.subscriber) {
            new lambda.Function(this, 'fn', {
                runtime: lambda.Runtime.NODEJS_14_X,
                code: lambda.Code.fromInline('console.error("hi")'),
                handler: 'index.handler',
                environment: {
                    TOPIC_ARN: props.siblingTopic?.topicArn ?? '',
                    QUEUE_URL: props.subscriber.queueUrl,
                },
            });
        }
    }
}
class MyTestStack extends core_1.Stack {
    constructor(scope, id) {
        super(scope, id);
        const queue = new sqs.Queue(this, 'SubscriberQueue');
        new MyNestedStack(this, 'NestedStack1', { topicCount: 3, topicNamePrefix: 'Prefix1', subscriber: queue });
        new MyNestedStack(this, 'NestedStack2', { topicCount: 2, topicNamePrefix: 'Prefix2', description: 'This is secound nested stack.' });
    }
}
const app = new core_1.App();
new MyTestStack(app, 'nested-stacks-test');
app.synth();
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW50ZWcubmVzdGVkLXN0YWNrLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiaW50ZWcubmVzdGVkLXN0YWNrLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7O0FBQUEsOENBQThDO0FBQzlDLHdDQUF3QztBQUN4QyxvRUFBb0U7QUFDcEUsd0NBQXdDO0FBQ3hDLHdDQUFzRTtBQWF0RSxNQUFNLGFBQWMsU0FBUSxrQkFBVztJQUNyQyxZQUFZLEtBQWdCLEVBQUUsRUFBVSxFQUFFLEtBQXlCO1FBQ2pFLE1BQU0sd0JBQXdCLEdBQUcsaUJBQWlCLENBQUM7UUFFbkQsS0FBSyxDQUFDLEtBQUssRUFBRSxFQUFFLEVBQUU7WUFDZixVQUFVLEVBQUU7Z0JBQ1YsQ0FBQyx3QkFBd0IsQ0FBQyxFQUFFLEtBQUssQ0FBQyxlQUFlO2FBQ2xEO1lBQ0QsV0FBVyxFQUFFLEtBQUssQ0FBQyxXQUFXO1NBQy9CLENBQUMsQ0FBQztRQUVILE1BQU0sd0JBQXdCLEdBQUcsSUFBSSxtQkFBWSxDQUFDLElBQUksRUFBRSxpQkFBaUIsRUFBRSxFQUFFLElBQUksRUFBRSxRQUFRLEVBQUUsQ0FBQyxDQUFDO1FBRS9GLEtBQUssSUFBSSxDQUFDLEdBQUcsQ0FBQyxFQUFFLENBQUMsR0FBRyxLQUFLLENBQUMsVUFBVSxFQUFFLEVBQUUsQ0FBQyxFQUFFO1lBQ3pDLE1BQU0sS0FBSyxHQUFHLElBQUksR0FBRyxDQUFDLEtBQUssQ0FBQyxJQUFJLEVBQUUsU0FBUyxDQUFDLEVBQUUsRUFBRSxFQUFFLFdBQVcsRUFBRSxHQUFHLHdCQUF3QixDQUFDLGFBQWEsSUFBSSxDQUFDLEVBQUUsRUFBRSxDQUFDLENBQUM7WUFFbkgsK0VBQStFO1lBQy9FLHVGQUF1RjtZQUN2RixJQUFJLEtBQUssQ0FBQyxVQUFVLEVBQUU7Z0JBQ3BCLEtBQUssQ0FBQyxlQUFlLENBQUMsSUFBSSxpQkFBaUIsQ0FBQyxlQUFlLENBQUMsS0FBSyxDQUFDLFVBQVUsQ0FBQyxDQUFDLENBQUM7YUFDaEY7U0FDRjtRQUVELElBQUksS0FBSyxDQUFDLFVBQVUsRUFBRTtZQUNwQixJQUFJLE1BQU0sQ0FBQyxRQUFRLENBQUMsSUFBSSxFQUFFLElBQUksRUFBRTtnQkFDOUIsT0FBTyxFQUFFLE1BQU0sQ0FBQyxPQUFPLENBQUMsV0FBVztnQkFDbkMsSUFBSSxFQUFFLE1BQU0sQ0FBQyxJQUFJLENBQUMsVUFBVSxDQUFDLHFCQUFxQixDQUFDO2dCQUNuRCxPQUFPLEVBQUUsZUFBZTtnQkFDeEIsV0FBVyxFQUFFO29CQUNYLFNBQVMsRUFBRSxLQUFLLENBQUMsWUFBWSxFQUFFLFFBQVEsSUFBSSxFQUFFO29CQUM3QyxTQUFTLEVBQUUsS0FBSyxDQUFDLFVBQVUsQ0FBQyxRQUFRO2lCQUNyQzthQUNGLENBQUMsQ0FBQztTQUNKO0tBQ0Y7Q0FDRjtBQUVELE1BQU0sV0FBWSxTQUFRLFlBQUs7SUFDN0IsWUFBWSxLQUFnQixFQUFFLEVBQVU7UUFDdEMsS0FBSyxDQUFDLEtBQUssRUFBRSxFQUFFLENBQUMsQ0FBQztRQUVqQixNQUFNLEtBQUssR0FBRyxJQUFJLEdBQUcsQ0FBQyxLQUFLLENBQUMsSUFBSSxFQUFFLGlCQUFpQixDQUFDLENBQUM7UUFFckQsSUFBSSxhQUFhLENBQUMsSUFBSSxFQUFFLGNBQWMsRUFBRSxFQUFFLFVBQVUsRUFBRSxDQUFDLEVBQUUsZUFBZSxFQUFFLFNBQVMsRUFBRSxVQUFVLEVBQUUsS0FBSyxFQUFFLENBQUMsQ0FBQztRQUMxRyxJQUFJLGFBQWEsQ0FBQyxJQUFJLEVBQUUsY0FBYyxFQUFFLEVBQUUsVUFBVSxFQUFFLENBQUMsRUFBRSxlQUFlLEVBQUUsU0FBUyxFQUFFLFdBQVcsRUFBRSwrQkFBK0IsRUFBRSxDQUFDLENBQUM7S0FDdEk7Q0FDRjtBQUVELE1BQU0sR0FBRyxHQUFHLElBQUksVUFBRyxFQUFFLENBQUM7QUFDdEIsSUFBSSxXQUFXLENBQUMsR0FBRyxFQUFFLG9CQUFvQixDQUFDLENBQUM7QUFDM0MsR0FBRyxDQUFDLEtBQUssRUFBRSxDQUFDIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICogYXMgbGFtYmRhIGZyb20gJ0Bhd3MtY2RrL2F3cy1sYW1iZGEnO1xuaW1wb3J0ICogYXMgc25zIGZyb20gJ0Bhd3MtY2RrL2F3cy1zbnMnO1xuaW1wb3J0ICogYXMgc25zX3N1YnNjcmlwdGlvbnMgZnJvbSAnQGF3cy1jZGsvYXdzLXNucy1zdWJzY3JpcHRpb25zJztcbmltcG9ydCAqIGFzIHNxcyBmcm9tICdAYXdzLWNkay9hd3Mtc3FzJztcbmltcG9ydCB7IEFwcCwgQ2ZuUGFyYW1ldGVyLCBOZXN0ZWRTdGFjaywgU3RhY2sgfSBmcm9tICdAYXdzLWNkay9jb3JlJztcbmltcG9ydCB7IENvbnN0cnVjdCB9IGZyb20gJ2NvbnN0cnVjdHMnO1xuXG4vKiBlc2xpbnQtZGlzYWJsZSBAYXdzLWNkay9uby1jb3JlLWNvbnN0cnVjdCAqL1xuXG5pbnRlcmZhY2UgTXlOZXN0ZWRTdGFja1Byb3BzIHtcbiAgcmVhZG9ubHkgc3Vic2NyaWJlcj86IHNxcy5RdWV1ZTtcbiAgcmVhZG9ubHkgc2libGluZ1RvcGljPzogc25zLlRvcGljOyAvLyBhIHRvcGljIGRlZmluZWQgaW4gYSBzaWJsaW5nIG5lc3RlZCBzdGFja1xuICByZWFkb25seSB0b3BpY0NvdW50OiBudW1iZXI7XG4gIHJlYWRvbmx5IHRvcGljTmFtZVByZWZpeDogc3RyaW5nO1xuICByZWFkb25seSBkZXNjcmlwdGlvbj86IHN0cmluZztcbn1cblxuY2xhc3MgTXlOZXN0ZWRTdGFjayBleHRlbmRzIE5lc3RlZFN0YWNrIHtcbiAgY29uc3RydWN0b3Ioc2NvcGU6IENvbnN0cnVjdCwgaWQ6IHN0cmluZywgcHJvcHM6IE15TmVzdGVkU3RhY2tQcm9wcykge1xuICAgIGNvbnN0IHRvcGljTmFtZVByZWZpeExvZ2ljYWxJZCA9ICdUb3BpY05hbWVQcmVmaXgnO1xuXG4gICAgc3VwZXIoc2NvcGUsIGlkLCB7XG4gICAgICBwYXJhbWV0ZXJzOiB7XG4gICAgICAgIFt0b3BpY05hbWVQcmVmaXhMb2dpY2FsSWRdOiBwcm9wcy50b3BpY05hbWVQcmVmaXgsIC8vIHBhc3MgaW4gYSBwYXJhbWV0ZXIgdG8gdGhlIG5lc3RlZCBzdGFja1xuICAgICAgfSxcbiAgICAgIGRlc2NyaXB0aW9uOiBwcm9wcy5kZXNjcmlwdGlvbixcbiAgICB9KTtcblxuICAgIGNvbnN0IHRvcGljTmFtZVByZWZpeFBhcmFtZXRlciA9IG5ldyBDZm5QYXJhbWV0ZXIodGhpcywgJ1RvcGljTmFtZVByZWZpeCcsIHsgdHlwZTogJ1N0cmluZycgfSk7XG5cbiAgICBmb3IgKGxldCBpID0gMDsgaSA8IHByb3BzLnRvcGljQ291bnQ7ICsraSkge1xuICAgICAgY29uc3QgdG9waWMgPSBuZXcgc25zLlRvcGljKHRoaXMsIGB0b3BpYy0ke2l9YCwgeyBkaXNwbGF5TmFtZTogYCR7dG9waWNOYW1lUHJlZml4UGFyYW1ldGVyLnZhbHVlQXNTdHJpbmd9LSR7aX1gIH0pO1xuXG4gICAgICAvLyBzaW5jZSB0aGUgc3Vic2NyaXB0aW9uIHJlc291cmNlcyBhcmUgZGVmaW5lZCBpbiB0aGUgc3Vic2NyaWJlcidzIHN0YWNrLCB0aGlzXG4gICAgICAvLyB3aWxsIGFkZCBhbiBTTlMgc3Vic2NyaXB0aW9uIHJlc291cmNlIHRvIHRoZSBwYXJlbnQgc3RhY2sgdGhhdCByZWZlcmVuY2UgdGhpcyB0b3BpYy5cbiAgICAgIGlmIChwcm9wcy5zdWJzY3JpYmVyKSB7XG4gICAgICAgIHRvcGljLmFkZFN1YnNjcmlwdGlvbihuZXcgc25zX3N1YnNjcmlwdGlvbnMuU3FzU3Vic2NyaXB0aW9uKHByb3BzLnN1YnNjcmliZXIpKTtcbiAgICAgIH1cbiAgICB9XG5cbiAgICBpZiAocHJvcHMuc3Vic2NyaWJlcikge1xuICAgICAgbmV3IGxhbWJkYS5GdW5jdGlvbih0aGlzLCAnZm4nLCB7XG4gICAgICAgIHJ1bnRpbWU6IGxhbWJkYS5SdW50aW1lLk5PREVKU18xNF9YLFxuICAgICAgICBjb2RlOiBsYW1iZGEuQ29kZS5mcm9tSW5saW5lKCdjb25zb2xlLmVycm9yKFwiaGlcIiknKSxcbiAgICAgICAgaGFuZGxlcjogJ2luZGV4LmhhbmRsZXInLFxuICAgICAgICBlbnZpcm9ubWVudDoge1xuICAgICAgICAgIFRPUElDX0FSTjogcHJvcHMuc2libGluZ1RvcGljPy50b3BpY0FybiA/PyAnJyxcbiAgICAgICAgICBRVUVVRV9VUkw6IHByb3BzLnN1YnNjcmliZXIucXVldWVVcmwsIC8vIG5lc3RlZCBzdGFjayByZWZlcmVuY2VzIGEgcmVzb3VyY2UgaW4gdGhlIHBhcmVudFxuICAgICAgICB9LFxuICAgICAgfSk7XG4gICAgfVxuICB9XG59XG5cbmNsYXNzIE15VGVzdFN0YWNrIGV4dGVuZHMgU3RhY2sge1xuICBjb25zdHJ1Y3RvcihzY29wZTogQ29uc3RydWN0LCBpZDogc3RyaW5nKSB7XG4gICAgc3VwZXIoc2NvcGUsIGlkKTtcblxuICAgIGNvbnN0IHF1ZXVlID0gbmV3IHNxcy5RdWV1ZSh0aGlzLCAnU3Vic2NyaWJlclF1ZXVlJyk7XG5cbiAgICBuZXcgTXlOZXN0ZWRTdGFjayh0aGlzLCAnTmVzdGVkU3RhY2sxJywgeyB0b3BpY0NvdW50OiAzLCB0b3BpY05hbWVQcmVmaXg6ICdQcmVmaXgxJywgc3Vic2NyaWJlcjogcXVldWUgfSk7XG4gICAgbmV3IE15TmVzdGVkU3RhY2sodGhpcywgJ05lc3RlZFN0YWNrMicsIHsgdG9waWNDb3VudDogMiwgdG9waWNOYW1lUHJlZml4OiAnUHJlZml4MicsIGRlc2NyaXB0aW9uOiAnVGhpcyBpcyBzZWNvdW5kIG5lc3RlZCBzdGFjay4nIH0pO1xuICB9XG59XG5cbmNvbnN0IGFwcCA9IG5ldyBBcHAoKTtcbm5ldyBNeVRlc3RTdGFjayhhcHAsICduZXN0ZWQtc3RhY2tzLXRlc3QnKTtcbmFwcC5zeW50aCgpO1xuIl19