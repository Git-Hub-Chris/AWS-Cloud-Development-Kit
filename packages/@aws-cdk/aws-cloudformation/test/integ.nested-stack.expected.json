{
  "Stack": {
    "Resources": {
      "SubscriberQueueC193DC66": {
        "Type": "AWS::SQS::Queue"
      },
      "SubscriberQueuePolicy25A0799E": {
        "Type": "AWS::SQS::QueuePolicy",
        "Properties": {
          "PolicyDocument": {
            "Statement": [
              {
                "Action": "sqs:SendMessage",
                "Condition": {
                  "ArnEquals": {
                    "aws:SourceArn": {
                      "Fn::GetAtt": [
                        "NestedStack1NestedStackNestedStack1NestedStackResource7E64AD52",
                        "Outputs.nestedstackstestNestedStack1topic02C2D1254Ref"
                      ]
                    }
                  }
                },
                "Effect": "Allow",
                "Principal": {
                  "Service": "sns.amazonaws.com"
                },
                "Resource": {
                  "Fn::GetAtt": [
                    "SubscriberQueueC193DC66",
                    "Arn"
                  ]
                }
              },
              {
                "Action": "sqs:SendMessage",
                "Condition": {
                  "ArnEquals": {
                    "aws:SourceArn": {
                      "Fn::GetAtt": [
                        "NestedStack1NestedStackNestedStack1NestedStackResource7E64AD52",
                        "Outputs.nestedstackstestNestedStack1topic1474E5389Ref"
                      ]
                    }
                  }
                },
                "Effect": "Allow",
                "Principal": {
                  "Service": "sns.amazonaws.com"
                },
                "Resource": {
                  "Fn::GetAtt": [
                    "SubscriberQueueC193DC66",
                    "Arn"
                  ]
                }
              },
              {
                "Action": "sqs:SendMessage",
                "Condition": {
                  "ArnEquals": {
                    "aws:SourceArn": {
                      "Fn::GetAtt": [
                        "NestedStack1NestedStackNestedStack1NestedStackResource7E64AD52",
                        "Outputs.nestedstackstestNestedStack1topic22C710DC4Ref"
                      ]
                    }
                  }
                },
                "Effect": "Allow",
                "Principal": {
                  "Service": "sns.amazonaws.com"
                },
                "Resource": {
                  "Fn::GetAtt": [
                    "SubscriberQueueC193DC66",
                    "Arn"
                  ]
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "Queues": [
            {
              "Ref": "SubscriberQueueC193DC66"
            }
          ]
        }
      },
      "SubscriberQueuenestedstackstestNestedStack1topic089C5EB1396F65087": {
        "Type": "AWS::SNS::Subscription",
        "Properties": {
          "Protocol": "sqs",
          "TopicArn": {
            "Fn::GetAtt": [
              "NestedStack1NestedStackNestedStack1NestedStackResource7E64AD52",
              "Outputs.nestedstackstestNestedStack1topic02C2D1254Ref"
            ]
          },
          "Endpoint": {
            "Fn::GetAtt": [
              "SubscriberQueueC193DC66",
              "Arn"
            ]
          }
        }
      },
      "SubscriberQueuenestedstackstestNestedStack1topic1150E1A929A2C267E": {
        "Type": "AWS::SNS::Subscription",
        "Properties": {
          "Protocol": "sqs",
          "TopicArn": {
            "Fn::GetAtt": [
              "NestedStack1NestedStackNestedStack1NestedStackResource7E64AD52",
              "Outputs.nestedstackstestNestedStack1topic1474E5389Ref"
            ]
          },
          "Endpoint": {
            "Fn::GetAtt": [
              "SubscriberQueueC193DC66",
              "Arn"
            ]
          }
        }
      },
      "SubscriberQueuenestedstackstestNestedStack1topic209B8719858511914": {
        "Type": "AWS::SNS::Subscription",
        "Properties": {
          "Protocol": "sqs",
          "TopicArn": {
            "Fn::GetAtt": [
              "NestedStack1NestedStackNestedStack1NestedStackResource7E64AD52",
              "Outputs.nestedstackstestNestedStack1topic22C710DC4Ref"
            ]
          },
          "Endpoint": {
            "Fn::GetAtt": [
              "SubscriberQueueC193DC66",
              "Arn"
            ]
          }
        }
      },
      "NestedStack1NestedStackNestedStack1NestedStackResource7E64AD52": {
        "Type": "AWS::CloudFormation::Stack",
        "Properties": {
          "TemplateURL": {
            "Fn::Join": [
              "",
              [
                "https://s3.",
                {
                  "Ref": "AWS::Region"
                },
                ".",
                {
                  "Ref": "AWS::URLSuffix"
                },
                "/",
                {
                  "Ref": "AssetParametersf3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8eS3Bucket3945C047"
                },
                "/",
                {
                  "Fn::Select": [
                    0,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParametersf3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8eS3VersionKey2EE8C10A"
                        }
                      ]
                    }
                  ]
                },
                {
                  "Fn::Select": [
                    1,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParametersf3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8eS3VersionKey2EE8C10A"
                        }
                      ]
                    }
                  ]
                }
              ]
            ]
          },
          "Parameters": {
            "TopicNamePrefix": "Prefix1",
            "referencetonestedstackstestSubscriberQueue39409787Ref": {
              "Ref": "SubscriberQueueC193DC66"
            }
          }
        }
      },
      "NestedStack2NestedStackNestedStack2NestedStackResourceFDF82E43": {
        "Type": "AWS::CloudFormation::Stack",
        "Properties": {
          "TemplateURL": {
            "Fn::Join": [
              "",
              [
                "https://s3.",
                {
                  "Ref": "AWS::Region"
                },
                ".",
                {
                  "Ref": "AWS::URLSuffix"
                },
                "/",
                {
                  "Ref": "AssetParameters45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904aS3Bucket107FF573"
                },
                "/",
                {
                  "Fn::Select": [
                    0,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParameters45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904aS3VersionKeyC63A1F96"
                        }
                      ]
                    }
                  ]
                },
                {
                  "Fn::Select": [
                    1,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParameters45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904aS3VersionKeyC63A1F96"
                        }
                      ]
                    }
                  ]
                }
              ]
            ]
          },
          "Parameters": {
            "TopicNamePrefix": "Prefix2"
          }
        }
      }
    },
    "Parameters": {
      "AssetParametersf3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8eS3Bucket3945C047": {
        "Type": "String",
        "Description": "S3 bucket for asset \"f3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8e\""
      },
      "AssetParametersf3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8eS3VersionKey2EE8C10A": {
        "Type": "String",
        "Description": "S3 key for asset version \"f3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8e\""
      },
      "AssetParametersf3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8eArtifactHash246B99B7": {
        "Type": "String",
        "Description": "Artifact hash for asset \"f3f09ec34d59b03270346926dc9298b8300df69c65edc0522507830dd4a97f8e\""
      },
      "AssetParameters45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904aS3Bucket107FF573": {
        "Type": "String",
        "Description": "S3 bucket for asset \"45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904a\""
      },
      "AssetParameters45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904aS3VersionKeyC63A1F96": {
        "Type": "String",
        "Description": "S3 key for asset version \"45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904a\""
      },
      "AssetParameters45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904aArtifactHashC28F299E": {
        "Type": "String",
        "Description": "Artifact hash for asset \"45243d26c80391394a05adecbb0afd6fff947bca4fe2e1b0b5f3e905d5ae904a\""
      }
    }
  },
  "NestedStacks": {
    "nested-stacks-test": [
      {
        "NestedStack1NestedStackNestedStack1NestedStackResource7E64AD52": {
          "Template": {
            "Metadata": {
              "aws:cdk:path": "nestedstackstestNestedStack1C289F6D5"
            },
            "Parameters": {
              "TopicNamePrefix": {
                "Type": "String"
              },
              "referencetonestedstackstestSubscriberQueue39409787Ref": {
                "Type": "String"
              }
            },
            "Resources": {
              "topic096865AFD": {
                "Type": "AWS::SNS::Topic",
                "Properties": {
                  "DisplayName": {
                    "Fn::Join": [
                      "",
                      [
                        {
                          "Ref": "TopicNamePrefix"
                        },
                        "-0"
                      ]
                    ]
                  }
                }
              },
              "topic119487C5C": {
                "Type": "AWS::SNS::Topic",
                "Properties": {
                  "DisplayName": {
                    "Fn::Join": [
                      "",
                      [
                        {
                          "Ref": "TopicNamePrefix"
                        },
                        "-1"
                      ]
                    ]
                  }
                }
              },
              "topic2EE13C5A5": {
                "Type": "AWS::SNS::Topic",
                "Properties": {
                  "DisplayName": {
                    "Fn::Join": [
                      "",
                      [
                        {
                          "Ref": "TopicNamePrefix"
                        },
                        "-2"
                      ]
                    ]
                  }
                }
              },
              "fnServiceRole5D180AFD": {
                "Type": "AWS::IAM::Role",
                "Properties": {
                  "AssumeRolePolicyDocument": {
                    "Statement": [
                      {
                        "Action": "sts:AssumeRole",
                        "Effect": "Allow",
                        "Principal": {
                          "Service": "lambda.amazonaws.com"
                        }
                      }
                    ],
                    "Version": "2012-10-17"
                  },
                  "ManagedPolicyArns": [
                    {
                      "Fn::Join": [
                        "",
                        [
                          "arn:",
                          {
                            "Ref": "AWS::Partition"
                          },
                          ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
                        ]
                      ]
                    }
                  ]
                }
              },
              "fn5FF616E3": {
                "Type": "AWS::Lambda::Function",
                "Properties": {
                  "Code": {
                    "ZipFile": "console.error(\"hi\")"
                  },
                  "Handler": "index.handler",
                  "Role": {
                    "Fn::GetAtt": [
                      "fnServiceRole5D180AFD",
                      "Arn"
                    ]
                  },
                  "Runtime": "nodejs10.x",
                  "Environment": {
                    "Variables": {
                      "TOPIC_ARN": "",
                      "QUEUE_URL": {
                        "Ref": "referencetonestedstackstestSubscriberQueue39409787Ref"
                      }
                    }
                  }
                },
                "DependsOn": [
                  "fnServiceRole5D180AFD"
                ]
              }
            },
            "Outputs": {
              "nestedstackstestNestedStack1topic02C2D1254Ref": {
                "Value": {
                  "Ref": "topic096865AFD"
                }
              },
              "nestedstackstestNestedStack1topic1474E5389Ref": {
                "Value": {
                  "Ref": "topic119487C5C"
                }
              },
              "nestedstackstestNestedStack1topic22C710DC4Ref": {
                "Value": {
                  "Ref": "topic2EE13C5A5"
                }
              }
            }
          }
        }
      },
      {
        "NestedStack2NestedStackNestedStack2NestedStackResourceFDF82E43": {
          "Template": {
            "Metadata": {
              "aws:cdk:path": "nestedstackstestNestedStack209C5218A"
            },
            "Parameters": {
              "TopicNamePrefix": {
                "Type": "String"
              }
            },
            "Resources": {
              "topic096865AFD": {
                "Type": "AWS::SNS::Topic",
                "Properties": {
                  "DisplayName": {
                    "Fn::Join": [
                      "",
                      [
                        {
                          "Ref": "TopicNamePrefix"
                        },
                        "-0"
                      ]
                    ]
                  }
                }
              },
              "topic119487C5C": {
                "Type": "AWS::SNS::Topic",
                "Properties": {
                  "DisplayName": {
                    "Fn::Join": [
                      "",
                      [
                        {
                          "Ref": "TopicNamePrefix"
                        },
                        "-1"
                      ]
                    ]
                  }
                }
              }
            }
          }
        }
      }
    ]
  }
}